python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -456,36 +452,101 @@ def __init__(self, n=1, normalize=False, **kwds):
             "minute",
             "second",
             "microsecond",
-            "millisecond" "nanosecond",
+            "millisecond",
+            "nanosecond",
         }
 
-        supported_kwargs = {"months"}
+        supported_kwargs = {
+            "years",
+            "months",
+            "weeks",
+            "days",
+            "hours",
+            "minutes",
+            "seconds",
+            "microseconds",
+            "nanoseconds",
+        }
+
+        super().__init__(n=n, normalize=normalize, **kwds)
+
+        kwds = self._combine_months_and_years(**kwds)
+        kwds = self._combine_timedeltas_to_nanos(**kwds)
 
         scalars = {}
         for k, v in kwds.items():
             if k in all_possible_kwargs:
                 # Months must be int16
-                dtype = "int16" if k == "months" else None
+                if k == "months":
+                    dtype = "int16"
+                elif k == "nanoseconds":
+                    dtype = "timedelta64[ns]"
+                else:
+                    dtype = None
                 scalars[k] = cudf.Scalar(v, dtype=dtype)
 
-        super().__init__(n=n, normalize=normalize, **kwds)
-
         wrong_kwargs = set(kwds.keys()).difference(supported_kwargs)
         if len(wrong_kwargs) > 0:
-            raise ValueError(
+            raise NotImplementedError(
                 f"Keyword arguments '{','.join(list(wrong_kwargs))}'"
                 " are not yet supported in cuDF DateOffsets"
             )
         self._scalars = _DateOffsetScalars(scalars)
 
-    def _generate_column(self, size, op):
-        months = self._scalars._gpu_scalars["months"]
+    def _combine_months_and_years(self, **kwargs):
+        kwargs["months"] = kwargs.pop("years", 0) * 12 + kwargs.pop(
+            "months", 0
+        )
+        return kwargs
+
+    def _combine_timedeltas_to_nanos(self, **kwargs):
+        weeks = kwargs.pop("weeks", 0)
+        days = kwargs.pop("days", 0) + 7 * weeks
+        hours = kwargs.pop("hours", 0) + days * 24
+        minutes = kwargs.pop("minutes", 0) + hours * 60
+        seconds = kwargs.pop("seconds", 0) + minutes * 60
+        milliseconds = kwargs.pop("milliseconds", 0) + seconds * 1000
+        microseconds = kwargs.pop("microseconds", 0) + milliseconds * 1000
+        nanoseconds = kwargs.pop("nanoseconds", 0) + microseconds * 1000
+        kwargs["nanoseconds"] = nanoseconds
+        return kwargs
```

Comment:
This has huge potential for overflowing the underlying `int64`. What resolutions do we support in cuDF / libcudf? Should we more strategically use those resolutions?

Reply:
Had a chat with @shwina , we'll move to returning the least common denominator of resolutions that we support.